### PR TITLE
Ref import procedure

### PR DESCRIPTION
**Pull request by Overtorment**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9z394bph0n7q3gwb1gm7593prc

[download apk](http://filestorage.bluewallet.io:1488/ref-import-procedure-1616698080.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/w94hvp3xr8kjyhqqp57nqtvkqr

[download apk](http://filestorage.bluewallet.io:1488/ref-import-procedure-1616701821.apk)

**Comment by jaonoctus** [comment]
Tested ACK.

## Unit tests

```
PASS  tests/unit/hd-aezeed.test.js (18.387 s)
  HDAezeedWallet
    ✓ can import mnemonics and generate addresses and WIFs (14653 ms)
    ✓ can sign and verify messages (2471 ms)
    
PASS  tests/unit/hd-legacy-electrum-seed-p2pkh-wallet.test.js
  HDLegacyElectrumSeedP2PKHWallet
    ✓ wont accept BIP39 seed (4 ms)
    ✓ wont accept electrum seed, but SEGWIT seed (1 ms)
    ✓ can import mnemonics and generate addresses and WIFs (27 ms)

PASS  tests/unit/hd-segwit-electrum-seed-p2wpkh-wallet.test.js
  HDSegwitElectrumSeedP2WPKHWallet
    ✓ wont accept BIP39 seed (3 ms)
    ✓ wont accept electrum seed, but STANDARD p2pkh seed (1 ms)
    ✓ can import mnemonics and generate addresses and WIFs (27 ms)
```

## Integration tests
```
 PASS  tests/integration/import.test.js (19.423 s)
  import procedure
    ✓ can import BIP84 (1541 ms)
    ✓ can import Legacy (457 ms)
    ✓ can import Legacy P2SH Segwit (654 ms)
    ✓ can import Legacy Bech32 Segwit (450 ms)
    ✓ can import BIP44 (719 ms)
    ✓ can import BIP49 (470 ms)
    ✓ can import BIP49 (955 ms)
    ✓ can import HD Legacy Electrum (BIP32 P2PKH) (18 ms)
    ✓ can import BreadWallet (954 ms)
    ✓ can import HD Electrum (BIP32 P2WPKH) (24 ms)
    ✓ can import AEZEED (1289 ms)
    ✓ importing empty BIP39 should yield BIP84 (933 ms)
    ✓ can import Legacy with uncompressed pubkey (445 ms)
    ✓ can import BIP38 encrypted backup (5692 ms)
    ✓ can import watch-only address (2451 ms)
```

File: tests/integration/import.test.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    assert.strictEqual(lastImportedWallet.getLabel(), 'Imported SegWit (P2SH)');
+  });
+
+  // todo:
+  it('can import Legacy Bech32 Segwit', async () => {
+    await WalletImport.processImportText('L1T6FfKpKHi8JE6eBKrsXkenw34d5FfFzJUZ6dLs2utxkSvsDfxZ');
+    assert.strictEqual(lastImportedWallet.type, SegwitBech32Wallet.type);
+    assert.strictEqual(lastImportedWallet.getAddress(), 'bc1q763rf54hzuncmf8dtlz558uqe4f247mq39rjvr');
+    assert.strictEqual(lastImportedWallet.getLabel(), 'Imported P2 WPKH');
+  });
```
Comment (jaonoctus) [inline tests/integration/import.test.js:94]: What is missing here?

Comment (Overtorment) [reply tests/integration/import.test.js:94]: forgot to remove `todo:`  after implementing test  >.<

**Review by angelanine3876 — APPROVED** [review]

**Review by limpbrains — APPROVED** [review]
`wasEverUsed` is fast, but may lead to more failed imports.
Maybe if it failed for all the wallets, we should try again with `fetchTransactions` ?

**Comment by Overtorment** [comment]
> `wasEverUsed` is fast, but may lead to more failed imports.
> Maybe if it failed for all the wallets, we should try again with `fetchTransactions` ?

that will slow things down when user wants to import blank seed on other device just to test how it is restored.
i recon `wasEverUsed` produce false negative if for some reason address /0/0 was skipped, and user started receiving on other addresses. a bit weird case. i suggest we start dealing with it if anyone will actually complain

**Review by marcosrdz — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**